p2p/message.go

```diff
@@ -58,15 +58,16 @@ func DecodeMessage(r io.Reader) (m Message, err error) {
 		return nil, err
 	}
 
-	sd := scale.Decoder{Reader: r}
-
 	switch msgType[0] {
-	case StatusMsg:
+	case StatusMsgType:
 		m = new(StatusMessage)
-		_, err = sd.Decode(m)
-	case BlockRequestMsg:
+		err = m.Decode(r)
+	case BlockRequestMsgType:
 		m = new(BlockRequestMessage)
-		_, err = sd.Decode(m)
+		err = m.Decode(r)
+	case BlockResponseMsgType:
+		m = new(BlockResponseMessage)
```

Comment:
Please write test for `BlockResponseMsgType`

Reply:
woops, thought I wrote one lol. added tests for everything :)

Reply:
@noot 

`go test -coverprofile cp.out`
`go tool cover -html=cp.out`

Indicates `(bm *BlockResponseMessage) Encode() ([]byte, error)` and `(bm *BlockResponseMessage) Decode(r io.Reader) error` are not covered by tests. 

The percentage sits below 70% for `p2p/message.go`. Please revise and improve coverage.